src/main.lua

```diff
@@ -1574,36 +1576,11 @@ end)
 
 -- TICK HANDLER - TODO: this may be better as a "Distribute" rewards handler instead of `Tick` tag
 addEventingHandler("distribute", utils.hasMatchingTag("Action", "Tick"), function(msg)
-	assert(msg.Timestamp, "Timestamp is required for a tick interaction")
 	local msgTimestamp = msg.Timestamp
-	-- tick and distribute rewards for every index between the last ticked epoch and the current epoch
-	local tickedRewardDistributions = {}
-	local totalTickedRewardsDistributed = 0
-
-	local function tickEpoch(timestamp, blockHeight, hashchain, msgId)
-		-- update demand factor if necessary
-		local demandFactor = demand.updateDemandFactor(timestamp)
-		-- distribute rewards for the epoch and increments stats for gateways, this closes the epoch if the timestamp is greater than the epochs required distribution timestamp
-		local distributedEpoch = epochs.distributeRewardsForEpoch(timestamp)
-		if distributedEpoch ~= nil and distributedEpoch.epochIndex ~= nil then
-			tickedRewardDistributions[tostring(distributedEpoch.epochIndex)] =
-				distributedEpoch.distributions.totalDistributedRewards
-			totalTickedRewardsDistributed = totalTickedRewardsDistributed
-				+ distributedEpoch.distributions.totalDistributedRewards
-		end
-		-- prune any gateway that has hit the failed 30 consecutive epoch threshold after the epoch has been distributed
-		local pruneGatewaysResult = gar.pruneGateways(timestamp, msgId)
-
-		-- now create the new epoch with the current message hashchain and block height
-		local newEpoch = epochs.createEpoch(timestamp, tonumber(blockHeight), hashchain)
-		return {
-			maybeEpoch = newEpoch,
-			maybeDemandFactor = demandFactor,
-			pruneGatewaysResult = pruneGatewaysResult,
-		}
-	end
 
 	local msgId = msg.Id
+	local blockHeight = tonumber(msg["Block-Height"])
```

Comment:
Should be a tag?

Reply:
And also already tonumbered?

Reply:
This is a top level attribute on all messages, similar to timestamp